source/reactor/unit_tests/EventLoopImplTest.cpp

```diff
@@ -0,0 +1,109 @@
+#include "reactor/EventHandlers.h"
+#include "reactor/EventLoopImpl.h"
+
+#include "TimeTools.h"
+
+#include <sys/timerfd.h>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using namespace u::reactor;
+
+using ::testing::AtLeast;
+using ::testing::Invoke;
+
+namespace {
+
+TEST(EventLoopImplTest, registerHandler_ShouldSucceed)
+{
+    EventLoopImpl loop;
+    EventHandlers handlers;
+
+    bool ok = loop.registerHandlers(0, 0, handlers);
+    loop.removeHandlers(0);
+
+    ASSERT_TRUE(ok);
+}
+
+TEST(EventLoopImplTest, removeHandler_ShouldSucceed)
+{
+    EventLoopImpl loop;
+    EventHandlers handlers;
+
+    loop.registerHandlers(0, 0, handlers);
+    bool ok = loop.removeHandlers(0);
+
+    ASSERT_TRUE(ok);
+}
+
+TEST(EventLoopImplTest, removeHandler_ShouldFailIfAlreadyRemoved)
+{
+    EventLoopImpl loop;
+    EventHandlers handlers;
+
+    loop.registerHandlers(0, 0, handlers);
+    loop.removeHandlers(0);
+    bool ok = loop.removeHandlers(0);
+
+    ASSERT_FALSE(ok);
+}
+
+TEST(EventLoopImplTest, removeHandler_ShouldFailIfNotAdded)
+{
+    EventLoopImpl loop;
+    EventHandlers handlers;
+
+    bool ok = loop.removeHandlers(0);
+
+    ASSERT_FALSE(ok);
+}
+
+TEST(EventLoopImplTest, runWithNoHandler_ShouldSucceed)
+{
+    EventLoopImpl loop;
+
+    bool ok = loop.run();
+
+    ASSERT_TRUE(ok);
+}
+
+TEST(EventLoopImplTest, terminate_ShouldSucceed)
+{
+    EventLoopImpl loop;
+    EventHandlers handlers;
+
+    handlers.onWrite = [&](int fd, EventLoop *loop) { loop->terminate(); };
```

Comment:
If you're not actually using any variable, IMO it's better to remove the reference:
```suggestion
    handlers.onWrite = [](int fd, EventLoop *loop) { loop->terminate(); };
```
To me, it feels like "const" in the sense that it tells the reader that whatever the function is doing is not going to leak into my local variables.